packages/@aws-cdk/aws-cognito/lib/user-pool-idps/apple.ts

```diff
@@ -0,0 +1,63 @@
+import { Construct } from 'constructs';
+import { CfnUserPoolIdentityProvider } from '../cognito.generated';
+import { UserPoolIdentityProviderProps } from './base';
+import { UserPoolIdentityProviderBase } from './private/user-pool-idp-base';
+
+/**
+ * Properties to initialize UserPoolAppleIdentityProvider
+ */
+export interface UserPoolIdentityProviderAppleProps extends UserPoolIdentityProviderProps {
+  /**
+   * The client id recognized by Apple APIs.
+   * @see https://developer.apple.com/documentation/sign_in_with_apple/clientconfigi/3230948-clientid
+   */
+  readonly clientId: string;
+  /**
+   * The teamId for Apple APIs to authenticate the client.
+   */
+  readonly teamId: string;
+  /**
+   * The keyId (which has to be later supplied as `privateKey`) for Apple APIs to authenticate the client.
```

Comment:
Does this mean keyid and private key are always the same?

Reply:
No it's not, but you are right, the comment is terrible. I will work on it :)